src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -229,14 +250,31 @@ protected int doWork() {
                 // First bring the simple metrics up to date
                 if (rec.getReadUnmappedFlag()) {
                     ++metrics.UNMAPPED_READS;
+                } else if(rec.isSecondaryOrSupplementary()) {
+                    ++metrics.SECONDARY_OR_SUPPLEMENTARY_READS_EXAMINED;
                 } else if (!rec.getReadPairedFlag() || rec.getMateUnmappedFlag()) {
                     ++metrics.UNPAIRED_READS_EXAMINED;
                 } else {
                     ++metrics.READ_PAIRS_EXAMINED; // will need to be divided by 2 at the end
                 }
 
+                // Now try and figure out the next duplicate index (if going by coordinate. if going by query name, only do this
+                // if the query name has changed.
+                final boolean needNextDuplicateIndex = recordInFileIndex > nextDuplicateIndex && //passed next duplicate
+                        ( sortOrder == SAMFileHeader.SortOrder.coordinate ||  //in coordinate sorted, always
+                                !rec.getReadName().equals(duplicateSetQueryName) && sortOrder == SAMFileHeader.SortOrder.queryname); //in querysorted if the query name has changed
```

Comment:
Please put parentheses when you have `( A || B && C)` to explicitly make it either `( A || (B && C))` or `((A || B) && C)`.